sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -806,8 +806,8 @@ class ParquetFilters(
               }
 
               override def keep(value: Binary): Boolean = {
-                value != null && UTF8String.fromBytes(value.getBytes).startsWith(
-                  UTF8String.fromBytes(strToBinary.getBytes))
+                value != null && UTF8String.fromBytes(value.getBytesUnsafe).startsWith(
+                  UTF8String.fromBytes(strToBinary.getBytesUnsafe))
```

Comment:
It's `ByteBufferBackedBinary`, and `isBackingBytesReused` is false. This change seems to be an improvement:
https://github.com/apache/parquet-java/blob/master/parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java#L473-L493